samples/hello-oboe/src/main/cpp/HelloOboeEngine.cpp

```diff
@@ -34,25 +34,30 @@
  * - Calculating the audio latency of the stream
  *
  */
-HelloOboeEngine::HelloOboeEngine(): mLatencyCallback(std::make_unique<LatencyTuningCallback>(*this)) {
+HelloOboeEngine::HelloOboeEngine()
+        : mLatencyCallback(std::make_unique<LatencyTuningCallback>(*this)) {
     start();
-    updateLatencyDetection();
 }
 
 double HelloOboeEngine::getCurrentOutputLatencyMillis() {
-    if (!mIsLatencyDetectionSupported) return -1;
+    if (!mIsLatencyDetectionSupported) return -1.0;
+
+    // Save local copy of shared pointer while we measure latency.
```

Comment:
Why is it necessary to copy the shared_ptr to a local var?

Reply:
We can delete the object if we only have one shared_ptr.  The code that reopens the stream will close and delete the old stream in mStream when it creates the new one. If we are referencing mStream then we may see it get deleted. By copying the shared_ptr we now hold a reference and can prevent the deletion. Another way to handle this would be to hold the mLock in this method. (I may have this all wrong. So I am glad you are asking.)

Reply:
I switched it to use a lock instead of localstream.

Reply:
> We can delete the object if we only have one shared_ptr. 

I getcha. Because we have 2 threads accessing `HelloOboeEngine` - the main thread and the "update latency timer"  thread we need to copy the `shared_ptr` to avoid the main thread from deleting the stream whilst it's being accessed by the "update latency timer" thread. Is that correct?